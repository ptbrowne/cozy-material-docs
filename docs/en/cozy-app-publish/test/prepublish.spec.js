/* eslint-env jest */
const prepublishLib = require('../lib/prepublish')

const downcloudSpy = jest.fn()
jest.doMock('../lib/hooks/pre/downcloud', () => downcloudSpy)

describe('Prepublish script', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const optionsMock = {
    appBuildUrl: 'http://example.mock',
    appSlug: 'cozy-mock',
    appType: 'webapp',
    appVersion: '0.0.1',
    registryUrl: 'http://registry.mock/',
    registryEditor: 'Cozy',
    registryToken: 'a5464f54e654c6546b54a56a'
  }

  it('generates sha256', async () => {
    await expect(prepublishLib(optionsMock)).resolves.toMatchSnapshot()
  })

  it('sanitize options from hook script', async () => {
    const options = {
      ...optionsMock,
      prepublishHook: './test/__mocks__/prepublish-unsanitized-hook'
    }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
  })

  it('check for undefined mandatory options', async () => {
    const options = {
      ...optionsMock,
      prepublishHook: './test/__mocks__/prepublish-missing-options-hook'
    }
    await expect(prepublishLib(options)).rejects.toThrowErrorMatchingSnapshot()
  })

  it('runs without errors with built-in hook', async () => {
    const options = { ...optionsMock, prepublishHook: 'downcloud' }
    await expect(prepublishLib(options)).resolves.toMatchSnapshot()
    expect(downcloudSpy).toHaveBeenCalled()
  })
})
