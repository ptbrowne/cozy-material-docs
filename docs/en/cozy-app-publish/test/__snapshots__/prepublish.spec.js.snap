// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prepublish script check for undefined mandatory options 1`] = `"appSlug cannot be undefined"`;

exports[`Prepublish script generates sha256 1`] = `
Object {
  "appBuildUrl": "http://example.mock",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
  "spaceName": undefined,
}
`;

exports[`Prepublish script runs without errors with built-in hook 1`] = `
Object {
  "appBuildUrl": "http://example.mock",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
  "spaceName": undefined,
}
`;

exports[`Prepublish script sanitize options from hook script 1`] = `
Object {
  "appBuildUrl": "http://example.mock",
  "appSlug": "cozy-mock",
  "appType": "webapp",
  "appVersion": "0.0.1",
  "buildCommit": undefined,
  "registryEditor": "Cozy",
  "registryToken": "a5464f54e654c6546b54a56a",
  "registryUrl": "http://registry.mock/",
  "sha256Sum": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
  "spaceName": undefined,
}
`;
